### DISCUSSION TITLE
Can you use internal Kubernetes DNS to receive traffic from other Kubernetes workloads?

### DISCUSSION THREAD

**Discussion by smeasures** [langfuse/langfuse#2344] [Support] [answered]
We have a self-hosted version of Langfuse running on Kubernetes. The setup is generally working fine, and the UI is accessible from outside the cluster via ingress via DNS, at langfuse._mycompany_.com.
We are running some Langchain/LangServe workloads on the same Kubernetes cluster, and we want to send our traces to LangFuse.
When we instrument Langfuse into our Langchain service, if we use the same DNS name of langfuse._mycompany_.com, it works OK, but I was hoping we would be able to use internal Kubernetes DNS, and set the Langfuse host to eg, langfuse-service.langfuse-namespace.svc.cluster.local.

Is this possible?
We are getting errors when we set it up this way, but they are not very clear on what the underlying issue is. I assume there is some issue with the fact that the host name doesn't match, but I just wanted to validate with the community before I keep trying to get it working.

**Comment by marcklingen** [comment, marked as answer]
The langfuse API does not check against any host name that you provide via env. Thus this should work as long as you make a proper request to the api. Can you expand on the errors, otherwise it’s difficult to help

**Reply by smeasures** [reply]
@marcklingen thank you for getting back to me.
With the confirmation that this should work, we kept investigating from other angles, and turns out we seemed to be using a version of Python that didn't seem to agree with Langfuse. We've reverted to a different version of Python and Langfuse is now working as intended, using the internal Kubernetes DNS service address.
Thanks for your help.
